tests/unit/utils/test_misc.py

```diff
@@ -132,3 +132,29 @@ def test_unflatten_returns_correct_shape() -> None:
 
     npt.assert_array_equal(tf.shape(unflat_y1), [2, 3, 4, 7])
     npt.assert_array_equal(tf.shape(unflat_y2), [2, 3, 4, 7, 11])
+
+
+@pytest.mark.parametrize(
+    "output_dims,expected_shape",
+    [
+        (1, [120]),
+        (2, [24, 5]),
+        (3, [6, 4, 5]),
+        (4, [2, 3, 4, 5]),
+    ],
+)
+def test_flatten_leading_dims_output_dims(output_dims: int, expected_shape: list[int]) -> None:
+    x_old = tf.random.uniform([2, 3, 4, 5])  # [2, 3, 4, 5]
+    flat_x_old, unflatten = flatten_leading_dims(x_old, output_dims=output_dims)  # [6, 4, 5]
```

Comment:
If you were restarting the tests for the PR, you could remove the shape comment as it doesn't apply to all test cases.